src/main/java/ro/hiringsystem/model/MailInformation.java

```diff
@@ -0,0 +1,34 @@
+package ro.hiringsystem.model;
+
+import lombok.*;
+
+import java.util.UUID;
+import java.util.function.BiPredicate;
+import java.util.function.Predicate;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class MailInformation {
+
+    private UUID id;
```

Comment:
Do we need an id for the mail? The mail itself is unique and can be identified by that. If this is meant to represent the userId it belongs too, then it is redundant as well and we can just store a list directly in the User object